### PR TITLE
Drivers rating page and function for sending via Telegram

### PR DESCRIPTION
**Pull request by bogdannoff**
Added new classes: DriversRatingMixin, DriversRating, UberDriversRating, BoltDriversRating, UklonDriversRating.

The method get_rating returns a structure with rating data. Data collects from models UklonPaymentsOrder, BoltPaymentsOrder, UberPaymentsOrder.

Added tests for these classes

Added view DriversRatingView and template for creating drivers rating page

Modified scripts\bot.py. Added function drivers_rating that sends rating via telegram

### CODE REVIEW

File: app/templates/app/drivers_rating.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                            <table class="table table-striped table-sm">
+                                <thead>
+                                    <tr>
+                                        <th class="col-1" scope="col">#</th>
+                                        <th class="col-6"scope="col">Driver</th>
+                                        <th class="col-1"scope="col">Trips</th>
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    {% for detail in x.rating %}
```
Comment (SergeyKutsko) [inline app/templates/app/drivers_rating.html:27]: @bogdannoff please use new model method `kassa()` to order by and show

File: scripts/driversrating.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import pendulum
+from django.db.models import Sum, F
+
+from app.models import UberPaymentsOrder, BoltPaymentsOrder, UklonPaymentsOrder
+
+
+class DriversRatingMixin:
+
+    def get_rating(self):
+        start_of_week = pendulum.now().start_of('week').subtract(days=3).start_of('week')
```
Comment (SergeyKutsko) [inline scripts/driversrating.py:10]: @bogdannoff  please reuse and DRY this code which alrady implemented in Uber Uklon and Bolt classes.

File: scripts/driversrating.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    def get_driver(self, item): return ''
+
+    def get_trips(self, item): return ''
+
+    def get_amount(self, item):
+        return item['total_amount__sum']
+
+    def start_of_week(self):
+        return pendulum.now().start_of('week').subtract(days=3).start_of('week')
```
Comment (SergeyKutsko) [inline scripts/driversrating.py:53]: @bogdannoff DRY code.

File: scripts/test_driversrating.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+from scripts.driversrating import UberDriversRating, BoltDriversRating, UklonDriversRating
+
+from django.test import TestCase
+
+from app.models import UberPaymentsOrder, BoltPaymentsOrder, UklonPaymentsOrder
+
+
+class UklonDriversRatingTest(TestCase):
+    start_of_week = datetime.datetime(2022, 10, 3, 0, 0, 0, tzinfo=datetime.timezone.utc)
```
Comment (SergeyKutsko) [inline scripts/test_driversrating.py:16]: @bogdannoff DRY code.

**Comment by SergeyKutsko** [comment]
@bogdannoff looks good, could you please add history of ratings so i can look per week?

**Comment by SergeyKutsko** [comment]
@bogdannoff please resolve conflicts and comments

File: scripts/bot.py [hunk 5]
```diff
@@ -69,6 +69,13 @@ def error_handler(update: object, context: CallbackContext) -> None:
     # Finally, send the message
     context.bot.send_message(chat_id=DEVELOPER_CHAT_ID, text=message, parse_mode=ParseMode.HTML)
 
+def drivers_rating(context):
+    text = 'Drivers Rating\n\n'
+    for fleet in DriversRatingMixin().get_rating():
+        text += fleet['fleet'] + '\n'
+        text += '\n'.join([f"{item['num']} {item['driver']} {item['trips']}" for item in fleet['rating']]) + '\n\n'
+    context.bot.send_message(chat_id=-828544906, text=text)
+
```
Comment (SergeyKutsko) [inline scripts/bot.py:78]: @bogdannoff lets run weekly rating at Monday 6AM

**Comment by SergeyKutsko** [comment]
@bogdannoff please fix comments and resolve conflicts

File: scripts/driversrating.py (hunk 2 above)
Comment (bogdannoff) [reply scripts/driversrating.py]: Done

File: scripts/driversrating.py (hunk 3 above)
Comment (bogdannoff) [reply scripts/driversrating.py]: Done

File: scripts/bot.py (hunk 5 above)
Comment (bogdannoff) [reply scripts/bot.py:80]: Done

**Comment by bogdannoff** [comment]
> 



> @bogdannoff looks good, could you please add history of ratings so i can look per week?

To receive history you have to use URL like /rating/?start=19-09-2022&end=19-09-2022

**Comment by SergeyKutsko** [comment]
@bogdannoff looks good. Here is your interview link https://calendly.com/universal_auto_internship/30min

**Pull request merged**